app/styles/spacing.scss

```diff
@@ -72,4 +72,16 @@ $space-map: (
   .width-#{$size} {
     width: $space;
   }
+
+  // Space
+  .space-y-#{$size} {
+    & > :not([hidden]) ~ :not([hidden]) {
+      margin-top: $space;
+    }
+  }
+  .space-x-#{$size} {
+    & > :not([hidden]) ~ :not([hidden]) {
```

Comment:
this selector selects all subsequent siblings of a child element that is not hidden, (given that the siblings are also not hidden). 

What this practically means, is that the first child doesn't get the margin - hence why it's margin-top for y and margin-left for x 🙏

Reply:
Interesting hadn't seen this as a solution before! Nice

Reply:
Guess where I stole it from ;)

Reply:
I don't understand this even with your explanation haha